ui/component/fileRenderFloating/view.jsx

```diff
@@ -226,69 +145,119 @@ export default function FileRenderFloating(props: Props) {
 
     // $FlowFixMe
     setFileViewerRect({ ...objectRect, windowOffset: window.pageYOffset });
-  }, [mainFilePlaying]);
 
-  useEffect(() => {
+    if (!mobilePlayerDimensions || mobilePlayerDimensions.height !== rect.height) {
+      doSetMobilePlayerDimensions({ height: rect.height, width: getScreenWidth() });
+    }
+  }, [doSetMobilePlayerDimensions, mainFilePlaying, mobilePlayerDimensions]);
+
+  const restoreToRelativePosition = React.useCallback(() => {
+    const SCROLL_BAR_PX = 12; // root: --body-scrollbar-width
+    const screenW = getScreenWidth() - SCROLL_BAR_PX;
+    const screenH = getScreenHeight();
+
+    const newX = Math.round(relativePosRef.current.x * screenW);
+    const newY = Math.round(relativePosRef.current.y * screenH);
+
+    setPosition(clampFloatingPlayerToScreen(newX, newY));
+  }, [setPosition]);
+
+  const clampToScreenOnResize = React.useCallback(
+    debounce(() => {
+      restoreToRelativePosition();
+    }, DEBOUNCE_WINDOW_RESIZE_HANDLER_MS),
+    []
+  );
+
+  // For playlists when pressing next/previous etc and switching players
+  function resetState() {
+    setCountdownCanceled(false);
+    setDoNavigate(false);
+    setPlayNext(true);
+  }
+
+  // ****************************************************************************
+  // EFFECTS
+  // ****************************************************************************
+
+  usePlayNext(
+    isFloating,
+    collectionId,
+    shouldPlayNext,
+    nextListUri,
+    previousListUri,
+    doNavigate,
+    doUriInitiatePlay,
+    resetState
+  );
+
+  React.useEffect(() => {
     if (playingUri && (playingUri.primaryUri || playingUri.uri)) {
       handleResize();
-      setCountdownCanceled(false);
     }
   }, [handleResize, playingUri, videoTheaterMode]);
 
-  useEffect(() => {
-    handleResize();
-    window.addEventListener('resize', handleResize);
-    onFullscreenChange(window, 'add', handleResize);
+  // Listen to main-window resizing and adjust the floating player position accordingly:
+  React.useEffect(() => {
+    if (isFloating) {
+      // Ensure player is within screen when 'isFloating' changes.
+      restoreToRelativePosition();
+    } else {
+      handleResize();
+    }
+
+    function onWindowResize() {
+      return isFloating ? clampToScreenOnResize() : handleResize();
+    }
+
+    window.addEventListener('resize', onWindowResize);
+    if (!isFloating) onFullscreenChange(window, 'add', handleResize);
 
     return () => {
-      window.removeEventListener('resize', handleResize);
-      onFullscreenChange(window, 'remove', handleResize);
+      window.removeEventListener('resize', onWindowResize);
+      if (!isFloating) onFullscreenChange(window, 'remove', handleResize);
     };
-  }, [handleResize]);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [clampToScreenOnResize, handleResize, isFloating]);
```

Comment:
On Safari, the video doesn't resize correctly when changing page zoom:
<img width="378" alt="image" src="https://user-images.githubusercontent.com/64950861/155316314-a23f7d0a-77db-4e35-9c1b-dec7e0dcc2bd.png">

Also, I think we should start entering reasons for why we are disabling lint and flow errors, so it's clear to the next person.  It is rare design to partially exclude something from the dep list (and I think the docs say it is usually wrong), but there's always a reason for everything.  It can be explained like this:
```
// eslint-disable-next-line react-hooks/exhaustive-deps, (blah... on-mount only, etc.)
```

Reply:
Hm .. seems like legacy issue.  Fix only if obvious, else can pass to Rave.  But the lint suggestion is still highly appreciated.  I'll start to do the same.